ipatests/test_ipaserver/test_adtrust_mockup.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2018  FreeIPA Project Contributors - see LICENSE file
+
+from __future__ import print_function
+import ipaserver.install.adtrust as adtr
+from ipaserver.install.adtrust import set_and_check_netbios_name
+from collections import namedtuple
+from unittest import TestCase
+from unittest.mock import MagicMock
+import sys
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+
+class ApiMockup:
```

Comment:
Nice! I love it!

In the long run, I'd like to see this more generalized and moved to a place that can be used by other tests. For now, keep it there.